src/granite/fields.cr

```diff
@@ -1,27 +1,64 @@
 require "json"
+require "uuid"
 
 module Granite::Fields
   alias SupportedArrayTypes = Array(String) | Array(Int16) | Array(Int32) | Array(Int64) | Array(Float32) | Array(Float64) | Array(Bool)
   alias Type = DB::Any | SupportedArrayTypes | UUID
   TIME_FORMAT_REGEX = /\d{4,}-\d{2,}-\d{2,}\s\d{2,}:\d{2,}:\d{2,}/
 
-  macro included
-    macro inherited
-      disable_granite_docs? CONTENT_FIELDS = {} of Nil => Nil
-      disable_granite_docs? FIELDS = {} of Nil => Nil
+  module ClassMethods
+    # All fields
+    def fields : Array(String)
+      {% begin %}
+        {% columns = @type.instance_vars.select { |ivar| ivar.annotation(Granite::Column) }.map(&.name.stringify) %}
+        {{columns.empty? ? "[] of String".id : columns}}
+      {% end %}
     end
+
+    # Fields minus the PK
+    def content_fields : Array(String)
+      {% begin %}
+        {% columns = @type.instance_vars.select { |ivar| (ann = ivar.annotation(Granite::Column)) && !ann[:primary] }.map(&.name.stringify) %}
+        {{columns.empty? ? "[] of String".id : columns}}
+      {% end %}
+    end
+  end
+
+  def content_values
+    parsed_params = [] of Type
+    {% for column in @type.instance_vars.select { |ivar| (ann = ivar.annotation(Granite::Column)) && !ann[:primary] } %}
+      {% ann = column.annotation(Granite::Column) %}
+      parsed_params << {% if ann[:converter] %} {{ann[:converter]}}.to_db {{column.name.id}} {% else %} {{column.name.id}} {% end %}
+    {% end %}
+    parsed_params
   end
 
   # specify the fields you want to define and types
   macro field(decl, **options)
     {% raise "The type of #{@type.name}##{decl.var} cannot be a Union.  The 'field' macro declares the type as nilable by default.  Use the 'field!' macro to declare a not nilable field." if decl.type.is_a? Union %}
-    {% CONTENT_FIELDS[decl.var] = options || {} of Nil => Nil %}
-    {% CONTENT_FIELDS[decl.var][:type] = decl.type %}
+    {% column_type = (options[:column_type] && !options[:column_type].nil?) ? options[:column_type] : nil %}
+    {% converter = (options[:converter] && !options[:converter].nil?) ? options[:converter] : nil %}
+    @[Granite::Column(column_type: {{column_type}}, converter: {{converter}})]
```

Comment:
The macro and annotation name should match.  `field` -> `@[Granite::Field()]` or `column` -> `@[Granite::Column()]`

Reply:
Kinda made this with the assumption that I would change it to be `column` in the future.  PR is still a WIP so thats on the TODO list.